backend/remote-state/http/backend.go

```diff
@@ -0,0 +1,334 @@
+package http
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/hashicorp/terraform/backend"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+// New Backend
+func New() backend.Backend {
+	s := &schema.Backend{
+		Schema: map[string]*schema.Schema{
+			"address": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "(Required) The address of the REST endpoint.",
+			},
+
+			"update_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "POST",
+				Description: "(Optional) HTTP method to use when updating state. Defaults to POST.",
+			},
+
+			"lock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the lock REST endpoint. Defaults to address.",
+			},
+
+			"lock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "LOCK",
+				Description: "(Optional) The HTTP method to use when locking. Defaults to LOCK.",
+			},
+
+			"unlock_address": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The address of the unlock REST endpoint. Defaults to address.",
+			},
+
+			"unlock_method": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "UNLOCK",
+				Description: "(Optional) The HTTP method to use when unlocking. Defaults to UNLOCK.",
+			},
+
+			"username": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The username for HTTP basic authentication.",
+			},
+
+			"password": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "(Optional) The password for HTTP basic authentication.",
+			},
+
+			"skip_cert_verification": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Default:     false,
+				Description: "(Optional) Whether to skip TLS verification. Defaults to false.",
+			},
+
+			"local_cert_ca_file": {
```

Comment:
why do you need both `local_cert_ca_file` and `local_cert_file`? If I understand your logic correctly, both of these cannot be set. So why two separate arguments? You can just have one `cert_file` argument. Also, these are named very similarly and it really confused me the first time I read this code.

Reply:
local_cert_ca_file is the CA that signed the certificate used by the REST API if you want to use ssl and don't want to skip_cert_validation since this is something that most of us don't want to. - here you authenticate only the REST API.

local_cert_file is the certificate used by terraform to authenticate with the REST API when mutual tls authentication is true. local_ca_file is the CA that signed the local_cert_file and local_key_file to decrypt. - here we authenticate terraform client with the REST API and also REST API with terraform. It's in the documentation https://github.com/wayfair/terraform/pull/1/files#diff-a1fb24cf00c253e06accbd7b9a7fadfeR66